### PR TITLE
Allows simple_animals to change intent

### PR DESCRIPTION
**Pull request by Kyep**
🆑 Kyep
add: Simple_animal mobs now get a HUD button that lets them switch between HELP and HARM intents.
/🆑

Why this is useful:
- Allows simple_animal mobs to block the passage of other mobs by switching to HARM intent.

HUD icons:
simplemob_hud_icons_row

Only help/harm are enabled currently for simplemobs, but the other icons are there in case anyone makes a mob that uses them in future.

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
This isn't the way to do it.

They should have a proper intent selector like every other mob has for changing intent.

This is special case, which should be avoided; if there's an existing system, *use that system*.

**Comment by Crazylemon64** [comment]
[act_intent selector code](https://github.com/ParadiseSS13/Paradise/blob/master/code/_onclick/hud/screen_objects.dm#L65)
[simple_animal hud creation code](https://github.com/ParadiseSS13/Paradise/blob/master/code/_onclick/hud/other_mobs.dm)

File: code/modules/mob/living/simple_animal/simple_animal.dm [hunk 1]
```diff
@@ -570,3 +578,32 @@
 
 /mob/living/simple_animal/SetEarDeaf()
 	return
+
+
+
+// Actions
+
+/datum/action/innate/simpleanimal/toggleintent
```
Comment (Crazylemon64) [inline code/modules/mob/living/simple_animal/simple_animal.dm]: [act_intent selector code](https://github.com/ParadiseSS13/Paradise/blob/master/code/_onclick/hud/screen_objects.dm#L65)
[simple_animal hud creation code](https://github.com/ParadiseSS13/Paradise/blob/master/code/_onclick/hud/other_mobs.dm)

**Comment by Tayyyyyyy** [comment]
Another simple animal UI improvement that could be helpful is fixing simple animals resting (doesn't seem to do anything, and is a source of confusion when people don't know why they can't move)

**Comment by Kyep** [comment]
@Fox-McCloud @Crazylemon64 changed to use screen/act_intent.

**Comment by Agameofscones** [comment]
If needed I could lend a hand here and mark intent icons for simple mobs

**Comment by Citinited** [comment]
Can you make different HUD icons for this? The robotic ones look, well, robotic, which doesn't really fit in to simple animals not being robots.

File: code/_onclick/hud/other_mobs.dm [hunk 2]
```diff
@@ -1,3 +1,15 @@
+/mob/living/simple_animal/create_mob_hud()
+	hud_used = new /datum/hud/simple(src)
+
+/datum/hud/simple/New(mob/user)
```
Comment (Crazylemon64) [inline code/_onclick/hud/other_mobs.dm]: You forgot to change this one

File: code/_onclick/hud/screen_objects.dm [hunk 3]
```diff
@@ -97,6 +94,11 @@
 	name = "run/walk toggle"
 	icon_state = "running"
 
+
+/obj/screen/act_intent/simple
```
Comment (Crazylemon64) [inline code/_onclick/hud/screen_objects.dm]: change this one as well

File: code/_onclick/hud/other_mobs.dm (hunk 2 above)
Comment (Kyep) [reply code/_onclick/hud/other_mobs.dm]: Change how?

File: code/_onclick/hud/screen_objects.dm (hunk 3 above)
Comment (Kyep) [reply code/_onclick/hud/screen_objects.dm]: Change how?

File: code/_onclick/hud/other_mobs.dm (hunk 2 above)
Comment (Crazylemon64) [reply code/_onclick/hud/other_mobs.dm]: Change it to `simple_animal`. A plain `simple` I see being easily confused for other systems.

**Comment by Kyep** [comment]
New icons from @Agameofscones are now in the PR, and shown in the PR description.

**Pull request merged**